components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2862,8 +2875,14 @@ private void patchIdpProperties(IdentityProvider identityProvider, String proper
         for (IdentityProviderProperty propertyDTO : propertyDTOS) {
             if (propertyName.equals(propertyDTO.getName())) {
                 propertyDTO.setValue(propertyValue);
+                return;
             }
         }
+
+        if (Constants.JWKS_URI.equals(propertyName)) {
+            throw handleException(Response.Status.NOT_FOUND, Constants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_IDP,
+                    "Cannot replace JWKS URI as it does not exist");
+        }
```

Comment:
This should be handled in a generic way for any missing property.

Reply:
Fixed